kong/plugins/ai-proxy/drivers/anthropic.lua

```diff
@@ -0,0 +1,250 @@
+local _M = {}
+
+-- imports
+local cjson = require("cjson.safe")
+local fmt = string.format
+local ai_shared = require("kong.plugins.ai-proxy.drivers.shared")
+local socket_url = require "socket.url"
+--
+
+local function kong_prompt_to_claude_prompt(prompt)
+  return fmt("Human: %s\n\nAssistant:", prompt)
+end
+
+local function kong_messages_to_claude_prompt(messages)
+  local buf = require("string.buffer").new()
```

Comment:
You'll want to put this line in the outer scope so that it is reused on each call rather than recreated from scratch:

```suggestion
-- maybe relocate this closer to the top of the file before your other functions
local buf = require("string.buffer").new()

local function kong_messages_to_claude_prompt(messages)
```

Reply:
^ with buffer reuse, just remember:

1. always `:reset()` before use
2. no yielding (network requests, `ngx.sleep()`, etc) any time between calling `:reset()` and `:get()`

(this code is following those rules already, just making sure it's known)

Reply:
But won't different threads cross over each buffer?

Reply:
If you'd like an extra measure of safety, it's also a common pattern to put mutable, long-lived variables used by a single function in a scope that is only accessible to said function:

```lua
local my_function
do
  -- this idiom makes it clear to readers of the code that the
  -- buffer's only intended use is within `my_function`. It is only
  -- reachable in this scope and therefore cannot be accidentally
  -- misused by any other function within the outer scope
  local buf = require("string.buffer").new()

  function my_function()
    buf:reset()
    buf:put("my string")
    return buf:get()
  end
end

function _M.foo()
  return my_function()
end
```

Reply:
I'm still lost on this one - looks like two requests timed exactly the same time, could write into the same buffer object, before one of them GET or FLUSH it.

Reply:
Discussed elsewhere, but I'll backtrack on this one. Let's leave it as-is (creating a new buffer for each invocation).

Reply:
Maybe move `require("string.buffer")` to the top of the file with the rest of your imports though:


```lua
local buffer = require "string.buffer"

-- [...]

local function kong_messages_to_claude_prompt(message)
  local buf = buffer.new()
  --- [...]
end
```